extensions/webjars-locator/deployment/src/main/java/io/quarkus/webjar/locator/deployment/WebJarLocatorStandaloneBuildStep.java

```diff
@@ -0,0 +1,58 @@
+package io.quarkus.webjar.locator.deployment;
+
+import java.util.Map;
+
+import org.webjars.WebJarAssetLocator;
+
+import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.vertx.http.deployment.RouteBuildItem;
+import io.quarkus.vertx.http.runtime.HttpBuildTimeConfig;
+import io.quarkus.webjar.locator.runtime.WebJarLocatorHandler;
+
+public class WebJarLocatorStandaloneBuildStep {
+
+    @BuildStep()
+    public void staticInit(Capabilities capabilities,
+            HttpBuildTimeConfig httpConfig,
+            BuildProducer<WebJarLocatorConfigBuildItem> standalone) throws Exception {
+        if (capabilities.isCapabilityPresent(Capabilities.SERVLET)) {
+            return;
+        }
+
+        WebJarAssetLocator webJarLocator = new WebJarAssetLocator();
+        Map<String, String> webjarNameToVersionMap = webJarLocator.getWebJars();
+
+        if (!webjarNameToVersionMap.isEmpty()) {
+            // The context path + the resources path
+            String rootPath = httpConfig.rootPath;
+            String webjarRootPath = (rootPath.endsWith("/")) ? rootPath + "webjars/" : rootPath + "/webjars/";
+            standalone.produce(new WebJarLocatorConfigBuildItem(webjarRootPath, webjarNameToVersionMap));
+        }
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void boot(
```

Comment:
I don't think you actually need WebJarLocatorConfigBuildItem, it looks like this could just be a single build step?

Reply:
Will there be any performance downside though? Or are all BuildStep (regardless if recorded or not) rerun at runtime? (the idea behind having it in two separate methods is to put the possibly computational expensive/classpath scanning at buildtime, and use the saved Map when we start the application)

Reply:
Build Steps are build-time only. What gets run at runtime are the recorders (that are associated with a build step).
So there is no reason to have different build steps for what you are doing.